### PR TITLE
[Flang] Fix for replacing loop uses in LoopVersioning pass

### PR DESCRIPTION
**Pull request by davemgreen**
The added test case has a loop that is versioned, which has a use of the loop in an if block after the loop. The current code replaces all uses of the loop with the new version If, but only if the parent blocks match. As far as I can see it should be safe to replace all the uses, then construct the result for the If with op.op.

### CODE REVIEW

**Comment by davemgreen** [comment]
Ping, thanks.

File: flang/test/Transforms/loop-versioning.fir [hunk 1]
```diff
@@ -1478,4 +1478,84 @@ func.func @sum1drebox(%arg0: !fir.box<!fir.array<?xf64>> {fir.bindc_name = "a"},
...
+}
+// CHECK-LABEL: func @minloc
+// CHECK: %[[V17:.*]] = fir.if %{{.*}} -> (i32) {
+// CHECK: %[[V27:.*]] = fir.do_loop
+// CHECK: fir.result %[[V27]] : i32
+// CHECK: } else {
+// CHECK: %[[V23:.*]] = fir.do_loop
+// CHECK: fir.result %[[V23]] : i32
+// CHECK: fir.if %{{.*}} {
+// CHECL: {{.*}} = arith.cmpi eq, %[[V17]], %c2147483647_i32
```
Comment (Leporacanthicus) [inline flang/test/Transforms/loop-versioning.fir:1557]: CHECL should be CHECK? [Is this a copy-paste from something I originally wrote - looks like something I'd do! :) If so, feel free to fix the original location too!]

File: flang/test/Transforms/loop-versioning.fir [hunk 2]
```diff
@@ -1478,4 +1478,84 @@ func.func @sum1drebox(%arg0: !fir.box<!fir.array<?xf64>> {fir.bindc_name = "a"},
...
+
+
```
Comment (Leporacanthicus) [inline flang/test/Transforms/loop-versioning.fir:1559]: Maybe just one or two empty lines at the end?

**Review by Leporacanthicus — COMMENTED** [review]
Overall looks OK to me. Found some minor things, see comments elsewhere.

File: flang/test/Transforms/loop-versioning.fir (hunk 1 above)
Comment (davemgreen) [reply flang/test/Transforms/loop-versioning.fir:1557]: I might be getting too used to auto-generated tests. Should be iupdated now

**Review by Leporacanthicus — APPROVED** [review]
LGTM

**Pull request merged**